src/i-shared-minter-mapping.ts

```diff
@@ -0,0 +1,164 @@
+import { BigInt, log, Address } from "@graphprotocol/graph-ts";
+import { Project } from "../generated/schema";
+
+import {
+  PricePerTokenInWeiUpdated,
+  ProjectCurrencyInfoUpdated,
+  ProjectMaxInvocationsLimitUpdated
+} from "../generated/ISharedMinterV0/ISharedMinterV0";
+
+import {
+  loadOrCreateMinter,
+  generateContractSpecificId,
+  loadOrCreateProjectMinterConfiguration
+} from "./helpers";
+
+///////////////////////////////////////////////////////////////////////////////
+// EVENT HANDLERS start here
+///////////////////////////////////////////////////////////////////////////////
+
+export function handlePricePerTokenInWeiUpdated(
+  event: PricePerTokenInWeiUpdated
+): void {
+  // attempt to load project, if it doesn't exist, log a warning and return
+  // @dev we don't support or allow minters to pre-configure projects that do
+  // not yet exist
+  const project = loadProjectByCoreAddressAndProjectNumber(
+    event.params._coreContract,
+    event.params._projectId
+  );
+  if (!project) {
+    log.warning("Project {} not found for core contract {}", [
+      event.params._projectId.toString(),
+      event.params._coreContract.toHexString()
+    ]);
+    return;
+  }
+
+  // load minter
+  const minter = loadOrCreateMinter(event.address, event.block.timestamp);
+
+  // load or create project minter configuration
+  const projectMinterConfig = loadOrCreateProjectMinterConfiguration(
+    project,
+    minter
+  );
+
+  projectMinterConfig.basePrice = event.params._pricePerTokenInWei;
+  projectMinterConfig.priceIsConfigured = true;
+  projectMinterConfig.save();
+
+  // only induce sync via updating project's updatedAt if the
+  // projectMinterConfig is the active minter configuration for the project
+  if (project.minterConfiguration == projectMinterConfig.id) {
+    project.updatedAt = event.block.timestamp;
+    project.save();
+  }
```

Comment:
Nit: we replicate this in multiple places, may be worth splitting into a smol util if we expect that this may drift otherwise.

Reply:
good call, we have this many places 😅 (including downstream PRs). I'll update and propagate downstream 👍

Reply:
added in [8551d76](https://github.com/ArtBlocks/artblocks-subgraph/pull/229/commits/8551d76335f6d613735ced5a15bc0a820f913edf), and will propagate to other PRs now